core/network/adapters/protobuf_block_request.hpp

```diff
@@ -28,19 +28,33 @@ namespace kagome::network {
         std::vector<uint8_t> &out,
         std::vector<uint8_t>::iterator loaded) {
       ::api::v1::BlockRequest msg;
-      msg.set_fields(LE_BE_SWAP32(t.fields.attributes.to_ulong()));
 
-      if (t.max) msg.set_max_blocks(*t.max);
-      if (t.to) msg.set_to_block(t.to->toString());
+      msg.set_fields(LE_BE_SWAP32(t.fields.attributes.to_ulong()));
 
-      msg.set_direction(static_cast<::api::v1::Direction>(t.direction));
       kagome::visit_in_place(
           t.from,
-          [&](const primitives::BlockHash &v) { msg.set_hash(v.toString()); },
-          [&](const primitives::BlockNumber &v) {
-            msg.set_number(std::to_string(v));
+          [&](const primitives::BlockHash &block_hash) {
+            msg.set_hash(block_hash.toString());
+          },
+          [&](const primitives::BlockNumber &block_number) {
+            uint32_t n = htole32(block_number);
+            msg.set_number(&n, sizeof(n));
           });
 
+      if (t.to) {
+        msg.set_to_block(t.to->toString());
+      } else {
+        msg.clear_to_block();
+      }
+
+      msg.set_direction(static_cast<::api::v1::Direction>(t.direction));
+
+      if (t.max) {
+        msg.set_max_blocks(*t.max);
+      } else {
+        msg.clear_max_blocks();
+      }
```

Comment:
comments would be precious

Reply:
Why? It is regular work with protobuf. Except that, clear_* calls is redundant.